Examples/Python/python/acts/_adapter.py

```diff
@@ -126,3 +126,9 @@ def _patch_detectors(m):
     for name, cls in inspect.getmembers(m, inspect.isclass):
         if name.endswith("Detector"):
             cls.create = _detector_create(cls)
+
+            cls.__enter__ = lambda self: self
+            if hasattr(cls, "drop"):
+                cls.__exit__ = lambda self, *args: self.drop()
+            else:
+                cls.__exit__ = lambda self, *args: None
```

Comment:
I'm wondering if it'd be cleaner to create a new class that inherits from the given one and replace the original one with it. That would avoid having to shuffle lambdas like this.

Reply:
Changed this back. I was unsure about how to handle this in the tests. I pass the tuple/context now instead of the individual detector things